tests/unit/sagemaker/workflow/test_processing_step.py

```diff
@@ -328,17 +328,27 @@ def test_processing_step_with_processor(pipeline_session, processing_input):
     }
 
 
-def test_processing_step_with_processor_and_step_args(pipeline_session, processing_input):
+@pytest.mark.parametrize(
+    "image_uri",
+    [
+        IMAGE_URI,
+        ParameterString(name="MyImage"),
+        ParameterString(name="MyImage", default_value="my-image-uri"),
+        Join(on="/", values=["docker", "my-fake-image"]),
+    ],
+)
+def test_processing_step_with_processor_and_step_args(
+    pipeline_session, processing_input, image_uri
+):
```

Comment:
This test is the fault case regarding the exclusive relationship between `processor` and `step_args`. 
Why we add the image_uri test case here?

Reply:
not following ...

Reply:
I meant usually we would separate the tests of happy case (e.g. parameterized image uri) and the failed case (e.g. the invalid input arguments of ProcessingStep) in different tests. But that's fine. It's just a Nit suggestion.